src/InstrumentationEngine/ProfilerManager.cpp

```diff
@@ -932,6 +922,12 @@ HRESULT CProfilerManager::Initialize(
         }
     }
 
+    if (FAILED(hr))
```

Comment:
This should go inside the "if (m_profilerCallbackHolder != nullptr)" so that it's only considered when the raw profiler is available. Also, consider just using IfFailRet on the Intialize calls instead of manually checking of failure, logging a custom error, and returning.

Reply:
Thanks